def personal_sum(*numbers):
    result = 0
    incorrect_data = 0

    for i in numbers:
        try:
            result += i
        except TypeError:
            incorrect_data += 1

    return result, incorrect_data


def calculate_average(numbers):
    try:
        res_sum = personal_sum(*numbers)
        if res_sum[0] == 0:
            return 0
        elif res_sum[1] > 0:
            print(f'Некорректных данных: {res_sum[1]}')
        return res_sum[0] / (len(numbers) - res_sum[1])

    except TypeError:
        print('В numbers записан некорректный тип данных')
        return None

    except ZeroDivisionError:
        return 0


print(f'Результат 1: {calculate_average("1, 2, 3")}\n')
print(f'Результат 2: {calculate_average([1, "Строка", 3, "Ещё Строка"])}\n')
print(f'Результат 3: {calculate_average(567)}\n')
print(f'Результат 4: {calculate_average([42, 15, 36, 13])}\n')
